src/controllers/expenseController.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+const {
+  getAll,
+  getById,
+  createOne,
+  deleteOne,
+  updateOne,
+} = require('../services/expensesServices');
+const { findAll } = require('../services/userServices');
+
+const get = (req, res) => {
+  const expenseList = getAll();
+  const { userId, categories, from, to } = req.query;
+
+  let filteredExpenses = expenseList;
+
+  if (categories && userId) {
+    filteredExpenses = expenseList
+      .filter(item => item.category === categories && item.userId === +userId);
+  } else if (userId) {
+    filteredExpenses = expenseList
+      .filter(item => item.userId === +userId);
+  }
+
+  if (from && to) {
+    filteredExpenses = filteredExpenses
+      .filter(item => item.spentAt > from && item.spentAt < to);
+  }
```

Comment:
Don't filter in the controller, any business logic should be placed in the service